### PR TITLE
`Exam mode`: Fix navbar exam detection and hide breadcrumbs appropriately

### PR DESCRIPTION
**Pull request by bassner**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

The navbar should hide a few elements (notifications, breadcrumbs ..) when an exam is running. However, at the moment there are a few problems with the related implementation:
- The template calls a method to check if the exam is active at multiple locations. This is a bad practice as the function is called over and over again to check for different results, causing continuous load
- The breadcrumbs only check if an exam is running when they are generated; however, this is not sufficient as the state of the exam can change over time, and without an navigation event, the breadcrumbs are not regenerated
- The current implementation does not take individual working times into account

### Description

- Reimplemented the check for an exam active to only populate a component field that can be consumed by the template that is updated by timeouts
- Hide the entire breadcrumb row if an exam is active instead of checking this while generating them
- Only hide elements for real exams (as we want do avoid distraction there). Do not do it for test runs or test exams

### Steps for Testing

Prerequisites:
- 1 Course with an upcoming Exam
- 1 Student registered for the exam, with a generated student exam so they can participate

1. Open the exam page as student before it starts. Notification icon and breadcrumbs should be visible.
2. Enter your name and click start. The countdown for the exam should appear as normal.
3. As soon as the countdown for the start of the exam runs out, you should be forwarded to the exercise overview. At the same time, breadcrumbs and the notification icon in the navbar etc should disappear.
4. Let the time of the exam run out. The hidden stuff should still be hidden.
5. Submit the exam OR let the submission time run out. In both cases the hidden things should reappear.

See the video below for how it could look like.

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| navbar.component.ts | 82% | 92% |

### Screenshots

https://user-images.githubusercontent.com/23171488/180625119-87f6835c-783c-4b27-b807-e9d36170caae.mov

### CODE REVIEW

**Review by nairaarnez — DISMISSED** [review]
Tested in TS1, submitting and letting the submission run out. Notification icon and breadcrumbs work as described

**Review by xlieA — DISMISSED** [review]
Tested on ts1. Everything works as described.

**Review by jbllhff — DISMISSED** [review]
Tested on ts2, works fine!

File: src/test/javascript/spec/component/admin/system-notification-management/system-notification-detail.component.spec.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ describe('SystemNotificationManagementDetailComponent', () => {
 
     beforeEach(() => {
         router = new MockRouter();
-        router.events = of({ id: 1, url: '' } as RouterEvent);
+        router.setUrl('');
```
Comment (JohannesStoehr) [inline src/test/javascript/spec/component/admin/system-notification-management/system-notification-detail.component.spec.ts:26]: There's a style issue in this file

File: src/test/javascript/spec/component/admin/system-notification-management/system-notification-management.component.spec.ts [hunk 2]
```diff
@@ -27,7 +27,7 @@ describe('SystemNotificationManagementComponent', () => {
 
     beforeEach(() => {
         router = new MockRouter();
-        router.events = of({ id: 1, url: '' } as RouterEvent);
+        router.setUrl('');
```
Comment (JohannesStoehr) [inline src/test/javascript/spec/component/admin/system-notification-management/system-notification-management.component.spec.ts:30]: Same as above

File: src/test/javascript/spec/service/navigation-util.service.spec.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ describe('Navigation Util Service', () => {
     let service: ArtemisNavigationUtilService;
 
     const router = new MockRouter();
-    router.events = of(new NavigationEnd(1, 'a', 'b'), new NavigationEnd(1, 'a', 'b'));
+    router.setUrl('a');
```
Comment (JohannesStoehr) [inline src/test/javascript/spec/service/navigation-util.service.spec.ts:13]: Style issues in the imports

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
Code looks good, but you have to fix the style issues

**Review by babyygemperor — DISMISSED** [review]
Tested on ts1, works as described

**Review by max-bergmann — DISMISSED** [review]
Tested on TS3. Works as expected

File: src/test/javascript/spec/component/admin/system-notification-management/system-notification-detail.component.spec.ts (hunk 1 above)
Comment (bassner) [reply src/test/javascript/spec/component/admin/system-notification-management/system-notification-detail.component.spec.ts:26]: See fdd1ac6 :)

File: src/test/javascript/spec/component/admin/system-notification-management/system-notification-management.component.spec.ts (hunk 2 above)
Comment (bassner) [reply src/test/javascript/spec/component/admin/system-notification-management/system-notification-management.component.spec.ts:30]: See fdd1ac6 :)

File: src/test/javascript/spec/service/navigation-util.service.spec.ts (hunk 3 above)
Comment (bassner) [reply src/test/javascript/spec/service/navigation-util.service.spec.ts:12]: See fdd1ac6 :)

**Comment by bassner** [comment]
Only changed imports in test files, so previous manual tests are still valid - I'd appreciate re-approvals, though :)

**Review by pantageepapa — APPROVED** [review]
Tested on ts2, and works as described :)

**Review by JohannesStoehr — APPROVED** [review]
Thanks for fixing the issues.
Code looks good

**Pull request merged**